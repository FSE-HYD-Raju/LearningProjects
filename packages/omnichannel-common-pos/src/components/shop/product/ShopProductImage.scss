@import '../../../variables';

.ShopProductImage {
	&-container {
		display: flex;
		flex-direction: column;
		width: 100%;
	}
	&-image {
		margin: auto;
		max-width: 100%;
		max-height: 12.5rem;
	}
	&-placeholder {
		background: gray("100");
		width: 100%;
		height: 100%;
	}

	&-shop-bundle-image,
	&-shop-bundle-image-empty {
		margin: auto;
		max-width: 150px;
	}

	&-shop-bundle-image-empty {
		background: #ddd;
		width: 30px;
		height: 50px;
		margin-top: 8px;
	}
}
