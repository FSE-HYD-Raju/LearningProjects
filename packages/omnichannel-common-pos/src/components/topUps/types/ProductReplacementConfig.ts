interface ProductReplacementConfig {
	individualId?: string;
	agreementId?: string;
	productId?: string;
	productOfferingId?: string;
	replaceType?: string;
	configuration?: object;
}
export { ProductReplacementConfig };
