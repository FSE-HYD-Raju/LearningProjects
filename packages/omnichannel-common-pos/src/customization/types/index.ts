import { CustomizationBootstrapType, GetStoreConfig } from "./CustomizationBootstrapType";
import { CustomizationClass } from "./CustomizationClass";
import { CustomizationFunction } from "./CustomizationFunction";
import { CustomizationPointsMapping } from "./CustomizationPointsMapping";

export {
	GetStoreConfig,
	CustomizationBootstrapType,
	CustomizationClass,
	CustomizationFunction,
	CustomizationPointsMapping
};
