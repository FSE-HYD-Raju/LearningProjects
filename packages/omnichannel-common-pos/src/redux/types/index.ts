import { ValidatePortInType } from "../../components/product/msisdn/MsisdnPortInSelector";
import { ConnectFluxReduxChannelType } from "./ConnectFluxReduxChannelType";
import { Disposers } from "./Disposers";
import { FluxAlt } from "./FluxAlt";
import { FluxReduxMap } from "./FluxReduxMap";
import { FluxStore } from "./FluxStore";
import { ReduxFluxActionMigration } from "./ReduxFluxActionMigration";
import { FluxWithActions } from "./fluxActions/FluxWithActions";

export {
	ConnectFluxReduxChannelType,
	Disposers,
	FluxAlt,
	FluxReduxMap,
	FluxStore,
	ReduxFluxActionMigration,
	FluxWithActions,
	ValidatePortInType
};

export * from "./utils/Omit";
export * from "./utils/HocResult";
export * from "./fluxActions";
export * from "../models/cms/cms.types";
export * from "../models/auth/auth.types";
export { CmsState } from "../models/cms/cms.reducers";
export { CmsAdminState } from "../models/cmsAdmin/cmsAdmin.reducers";
export * from "../models/auth/auth.types";
export * from "../models/service/service.types";
export * from "../models/support/support.types";
export * from "../models/comparison/comparison.types";
export * from "../models/consul/consul.types";
export * from "../models/currency/currency.types";
export * from "../models/eCare/changeSim/changeSim.types";
export * from "../models/delivery/delivery.types";
export * from "../models/digitalLife/digitalLife.types";
export * from "../models/eCare/activateSim/activateSim.types";
export * from "../models/eCare/suspension/suspension.types";
export * from "../models/feature/feature.types";
export * from "../models/lifecycle/lifecycle.types";
export * from "../models/organization/organization.types";
export * from "../models/productOfferingConfiguration/productOfferingConfiguration.types";
export * from "../models/schema/schema.types";
export * from "../models/location/location.types";
export * from "../models/eligibility/eligibility.types";
export * from "../models/productOfferings/productOfferings.types";
export * from "../models/user/user.types";
export * from "../models/basket/basket.types";
export * from "../models/customer/customer.types";
export * from "../models/sales/sales.types";
export * from "./SupportCases";
export * from "../models/versionInformation/versionInformation.types";
export * from "../models/msisdnSelection/msisdnSelection.types";
export * from "../models/portIn/portIn.types";
export * from "../models/marketingConsent/marketingConsent.types";
export * from "../models/category/category.types";
export * from "../models/b2cCheckout/b2cCheckout.types";
export { ProvinceAndCityState, Province, CityType } from "../models/provinceAndCity/provinceAndCity.types";

export * from "./routes/HasPath";
export * from "./routes/ApplicationRoute";
export * from "./HasSchema";
export * from "./HasFlux";
export * from "./HasId";
export * from "./HasPrices";
export * from "./BillingInterval";
export * from "./Cardinality";
export * from "./Characteristic";
export * from "./CharacteristicValue";
export * from "./CommercialEnrichments";
export * from "./Country";
export * from "./Links";
export * from "./FilterLabel";
export * from "./Gender";
export * from "./ImageType";
export * from "./Msisdn";
export * from "./MsisdnConfig";
export * from "./Price";
export * from "./SimplePrice";
export * from "./PriceRange";
export * from "./PriceType";
export * from "./PriceRangeBoundaries";
export * from "./ProductFilter";
export * from "./ProductOffering";
export * from "./ProductOfferingGroup";
export * from "./AlternateProductOffering";
export * from "./StockLevel";
export * from "./ProductTypes";
export * from "./ServiceUnit";
export * from "./Resource";
export * from "./SimCard";
export * from "./UnitOfMeasure";
export * from "./Allowance";
export * from "./ValidityPeriod";
export * from "./ValueRegulator";
export * from "./AutoCharacteristicConfigurationType";
export * from "./ChargingBalances";
export * from "./ChargingBalanceChanges";
export * from "./ProductsModifyConfig";
export * from "./Product";
export * from "./Agreement";
export * from "./ContactMediaType";
export * from "./EmailAddress";
export * from "./PhoneNumber";
export * from "./PostalAddress";
export * from "./Location";
export * from "./StreetAddressTokens";
export * from "./IssuingAuthority";
export * from "./Identification";
export * from "./CommonPerson";
export * from "./RelatedPerson";
export * from "./OrderLifeCycleStatus";
export * from "./Order";
export * from "./OrderCancel";
export * from "./OrderItem";
export * from "./OrderTotalsType";
export * from "./Person";
export * from "./MsisdnLifecycleStatus";
export * from "./Basket";
export * from "./payment/ContextualPayment";
export * from "./payment/ContextualPaymentMethod";
export * from "./payment/ContextualPaymentValidation";
export * from "./payment/CustomerPaymentMethod";
export * from "./payment/CustomerPaymentMethodType";
export * from "./payment/PaymentMethodsFiltersType";
export * from "./payment/CreditCard";
export * from "./payment/Role";
export * from "./BasketItem";
export * from "./SpecSubType";
export * from "./SpecType";
export * from "./TargetLifecycleStatus";
export * from "./BasketLifecycleStatus";
export * from "./BasketValidationAction";
export * from "./BasketValidationInformation";
export * from "./BasketProduct";
export * from "./PersonToPersonRelationship";
export * from "./CustomerCase";
export * from "./Category";
export * from "./SubCategory";
export * from "./ProductLifecycleStatus";
export * from "./UsageLimits";
export * from "./UsageCounters";
export * from "./Identifier";
export * from "./Notification";
export * from "./Document";
export * from "./BillingBalance";
export * from "./ShippingMethod";
export * from "./ChangeResourceInitialization";
export * from "./ChangeResource";
export * from "./Reason";
export * from "./MsisdnReservation";
export * from "./payment/PaymentStatus";
export * from "./payment/PaymentInvocation";
export * from "./Owner";
export * from "./CustomerInteraction";
export * from "./Initiator";
export * from "./InteractionDetails";
export * from "./AttachedDocument";
export * from "./Discount";
export * from "./Interval";
export * from "./BasketPaymentReceipt";
export * from "./PaginationData";
export * from "./PersonsOrderData";
export * from "./PersonsOrderState";
export * from "./OrderStatusFilter";
export * from "./SalesInfo";
export * from "./Shipment";
export * from "./AxiosResponse";
export * from "./CommercialEnrichmentName";
export * from "./UsageEvent";
export * from "./UsagePrice";
export * from "./NonMonetaryAmount";
export * from "./EventDetails";
export * from "./LifecycleChangeAction";
export * from "./Service";
export * from "./ServiceLifeCycleStatus";
export * from "./Specification";
export * from "./CustomerAccount";
export * from "./CustomerAccountLifecycleStatus";
export * from "./RecurringTopUpModelType";
export * from "../../components/priceComponent/model/PriceComponentPriceType";
export * from "./ErrorType";
export * from "./PostalAddressValidationErrorType";
export * from "./ProductModification";
export * from "./CallForwardingServiceModify";
export * from "./ProductOfferingFilter";
export * from "./ProductOfferingFilterMatcher";
export * from "./PortInDecision";
export * from "./ProductOfferingFilter";
export * from "./ProductOfferingFilterMatcher";
export * from "./DomainContext";
export * from "./ServiceModification";
export * from "./ResourceStocks";
export * from "./ResourceInventories";
export * from "./Inventory";
export * from "./SalesOrganizationRole";
export * from "./NominationCharacteristics";
export * from "./NominationPOCharacteristicsConfig";
export * from "./BasketItemsColumn";
export * from "./OrderItemsColumn";
export * from "./OrdersColumn";
export * from "./BasketsColumn";
export * from "./ConsumptionData";
export * from "./ProductsReplaceType";
export * from "./OmnichannelApiResponse";
export * from "./ProductConfiguration";
export * from "./TopUpCharacteristic";
export * from "./RecurringTopUpIntervalType";
export * from "./RecurringTopUpCharacteristicsValue";
export * from "./RecurringTopUpProductType";
export * from "./HasProduct";
export * from "./Store";
export * from "./QueryStatus";
export * from "./DeviceReservation";
export * from "./HasIntl";
export * from "./ProductUsageColumn";
export * from "./WorkforceAppointment";
