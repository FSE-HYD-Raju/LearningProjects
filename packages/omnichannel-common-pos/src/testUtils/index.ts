import ReactWidgetsTestUtils from "./ReactWidgetsTestUtils";
import * as MockDataMaker from "./mock";
const {
	shallowWithContextAndRouterProps,
	shallowWithContextAndLifecycle,
	deepShallowWithContextAndLifecycle,
	deepShallowWithContext,
	shallowWithContext,
	mountWithContextAndRouterProps,
	mountWithContext,
	attachWithContext,
	detach
} = require("./custom-mounters");
const TestUtils = require("./TestUtils");
const MockData = require("./MockData");
const SimpleDataMock = require("./mock/simpleDataMock");

export {
	MockData,
	ReactWidgetsTestUtils,
	TestUtils,
	SimpleDataMock,
	shallowWithContextAndRouterProps,
	shallowWithContextAndLifecycle,
	deepShallowWithContextAndLifecycle,
	deepShallowWithContext,
	shallowWithContext,
	mountWithContextAndRouterProps,
	mountWithContext,
	attachWithContext,
	detach,
	MockDataMaker
};

export * from "./MockRouterProps";

export default TestUtils;
