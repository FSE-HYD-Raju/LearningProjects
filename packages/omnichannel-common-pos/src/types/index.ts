import { ContextType, contextTypesValidationMap } from "./ContextType";

export {
	ContextType,
	contextTypesValidationMap
};
