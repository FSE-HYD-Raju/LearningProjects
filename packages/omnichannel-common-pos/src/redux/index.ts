"use strict";

import { default as actions, AppActions } from "./actions";
import { default as reducers, AppState } from "./reducers";
import { InterceptReducers } from "./reducers/reducerUtils";

import { default as DevTools } from "./DevTools";

import { connectFluxRedux, getDefaultFluxStoreMap } from "./connectFluxRedux";
import { configureStore, AppStore, StoreConfig } from "./store";
import * as actionTypes from "./actions";
import * as decorators from "./decorators";

import * as channelUtils from "./utils/channel";
export * from "./utils";
import * as Selectors from "./selectors";

import { initialState } from "./reducers/initialState";
import BaseService from "./services/BaseService";
export * from "./settings/core";
import { productToBasketItem as productToBasketItemTypes } from "./models/feature/feature.types";
// uses "configureStore" from ./store, should be imported after ./store
import * as ReduxTestUtils from "./utils/ReduxTestUtils";

export {
	decorators,
	Selectors,
	actions,
	reducers,
	DevTools,
	StoreConfig,
	InterceptReducers,
	AppStore,
	initialState,
	AppActions,
	AppState,
	connectFluxRedux,
	configureStore,
	getDefaultFluxStoreMap,
	ReduxTestUtils,
	channelUtils,
	actionTypes,
	BaseService,
	productToBasketItemTypes,
};

export * from "./actions/actionsPayload";
export * from "./services/Rest";
export * from "./services";
export * from "./types";
export * from "./decorators";
export * from "./models/msisdnSelection/msisdnSelection.actions";
export * from "./models/productOfferingConfiguration/productOfferingConfiguration.utils";
