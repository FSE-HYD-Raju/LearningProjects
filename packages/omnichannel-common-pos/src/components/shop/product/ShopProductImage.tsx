import { ProductOffering, CommercialEnrichments } from "../../../redux";
import cssns from "../../../utils/cssnsConfig";
import { get, isEmpty } from "lodash";
import classnames from "classnames";
import { FC } from "react";

const { React } = cssns("ShopProductImage");

export interface ShopProductImageProps {
	idx?: number;
	width?: string;
	height?: string;
	product: ProductOffering;
}

const ShopProductImage: FC<ShopProductImageProps> = props => {
	const { product } = props;

	const commercialEnrichments: Array<CommercialEnrichments> = get(product, "attributes.commercialEnrichments", []);
	const commercialEnrichment = commercialEnrichments.find((ce: CommercialEnrichments) => {
		return !isEmpty(ce.media);
	});

	const src = get(commercialEnrichment, "media.thumbnail-image");

	const customStyles = {
		width: props.width,
		height: props.height
	};

	const imageClasses = classnames({
		image: true,
		placeholder: !src
	});

	const combinedId = `ShopProductImage-${product.id}` + props.idx ? `-${props.idx}` : "";
	return (
		<div className="container">
			<img
				className={imageClasses}
				id={combinedId}
				src={src ? src : "data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"}
				style={props.width || props.height ? customStyles : {}}
			/>
		</div>
	);
};

export default ShopProductImage;
