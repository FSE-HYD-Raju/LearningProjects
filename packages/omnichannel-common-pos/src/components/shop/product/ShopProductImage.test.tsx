import * as React from "react";

import { mountWithContext, shallowWithContext } from "../../../testUtils";
import ShopProductImage, { ShopProductImageProps } from "./ShopProductImage";
import { ProductOffering } from "../../../redux/types";

describe("ShopProductImage", () => {
	const minimumProps: ShopProductImageProps = {
		product: {
			id: "ac7a8d35-8ff4-4d1b-ab92-ce64a257a849",
			attributes: {
				commercialEnrichments: []
			}
		} as any as ProductOffering
	};

	it("succeeds at shallow mount with minimum props", () => {
		const wrapper = shallowWithContext(
			<ShopProductImage {...minimumProps} />
		);
		expect(wrapper).toMatchSnapshot();
	});

	it("succeeds at deep mount with minimum props", () => {
		mountWithContext(<ShopProductImage {...minimumProps} />);
	});
});
