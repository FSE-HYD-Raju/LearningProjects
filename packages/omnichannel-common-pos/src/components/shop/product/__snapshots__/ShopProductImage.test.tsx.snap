// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShopProductImage succeeds at shallow mount with minimum props 1`] = `
<div
  className="ShopProductImage-container"
>
  <img
    className="ShopProductImage-image ShopProductImage-placeholder"
    id="-undefined"
    src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
    style={Object {}}
  />
</div>
`;
